src/main/kotlin/org/rust/ide/intentions/AddDerive.kt

```diff
@@ -0,0 +1,62 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.appendMetaToList
+import org.rust.lang.core.psi.util.parentOfType
+
+class AddDerive : PsiElementBaseIntentionAction()
+                , IntentionAction {
+    override fun getFamilyName() = "Add derive clause"
+    override fun getText() = "Add Derive Clause"
+    override fun startInWriteAction() = true
+
+    override fun invoke(project: Project, editor: Editor?, element: PsiElement) {
+        val item = element.parentOfType<RustItem>() ?: return
```

Comment:
Ah, I see this in `isAvailable` now. Hm, what about introducing `findStructOrEnum` helper function and using it here and in `isAvailable`? 

It should also allow to get rid of that `when`.